pydantic_ai_slim/pydantic_ai/agent.py

```diff
@@ -129,12 +137,19 @@ def __init__(
                 which checks for the necessary environment variables. Set this to `false`
                 to defer the evaluation until the first run. Useful if you want to
                 [override the model][pydantic_ai.Agent.override] for testing.
+            end_strategy: The strategy for handling tool calls that are requested alongside a final result.
+                "correct" means process all tool calls even after finding a final result.
+                "early" means stop processing other tool calls if a final result is found.
+                Defaults to "correct".
         """
         if model is None or defer_model_check:
             self.model = model
         else:
             self.model = models.infer_model(model)
 
+        if end_strategy not in ('early', 'correct'):
+            raise exceptions.UserError('end_strategy must be either "early" or "correct"')
```

Comment:
I think we can omit this, nothing bad will happen if you use another value. Also, we don't want to get into the habit of type checking everything.

And allow other values might be useful for future compatibility.